ante/fee.go

```diff
@@ -1,75 +1,95 @@
-// 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-// 	tmstrings "github.com/tendermint/tendermint/libs/strings"
-// )
-
-// const maxBypassMinFeeMsgGasUsage = uint64(200_000)
-
-// // MempoolFeeDecorator will check if the transaction's fee is at least as large
-// // as the local validator's minimum gasFee (defined in validator config).
-// //
-// // If fee is too low, decorator returns error and tx is rejected from mempool.
-// // Note this only applies when ctx.CheckTx = true. If fee is high enough or not
-// // CheckTx, then call next AnteHandler.
-// //
-// // CONTRACT: Tx must implement FeeTx to use MempoolFeeDecorator
-// type MempoolFeeDecorator struct {
-// 	BypassMinFeeMsgTypes []string
-// }
-
-// func NewMempoolFeeDecorator(bypassMsgTypes []string) MempoolFeeDecorator {
-// 	return MempoolFeeDecorator{
-// 		BypassMinFeeMsgTypes: bypassMsgTypes,
-// 	}
-// }
-
-// func (mfd MempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
-// 	feeTx, ok := tx.(sdk.FeeTx)
-// 	if !ok {
-// 		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
-// 	}
-
-// 	feeCoins := feeTx.GetFee()
-// 	gas := feeTx.GetGas()
-// 	msgs := feeTx.GetMsgs()
-
-// 	// Only check for minimum fees if the execution mode is CheckTx and the tx does
-// 	// not contain operator configured bypass messages. If the tx does contain
-// 	// operator configured bypass messages only, it's total gas must be less than
-// 	// or equal to a constant, otherwise minimum fees are checked to prevent spam.
-// 	if ctx.IsCheckTx() && !simulate && !(mfd.bypassMinFeeMsgs(msgs) && gas <= uint64(len(msgs))*maxBypassMinFeeMsgGasUsage) {
-// 		minGasPrices := ctx.MinGasPrices()
-// 		if !minGasPrices.IsZero() {
-// 			requiredFees := make(sdk.Coins, len(minGasPrices))
-
-// 			// Determine the required fees by multiplying each required minimum gas
-// 			// price by the gas limit, where fee = ceil(minGasPrice * gasLimit).
-// 			glDec := sdk.NewDec(int64(gas))
-// 			for i, gp := range minGasPrices {
-// 				fee := gp.Amount.Mul(glDec)
-// 				requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
-// 			}
-
-// 			if !feeCoins.IsAnyGTE(requiredFees) {
-// 				return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
-// 			}
-// 		}
-// 	}
-
-// 	return next(ctx, tx, simulate)
-// }
-
-// func (mfd MempoolFeeDecorator) bypassMinFeeMsgs(msgs []sdk.Msg) bool {
-// 	for _, msg := range msgs {
-// 		if tmstrings.StringInSlice(sdk.MsgTypeURL(msg), mfd.BypassMinFeeMsgTypes) {
-// 			continue
-// 		}
-
-// 		return false
-// 	}
-
-// 	return true
-// }
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	tmstrings "github.com/tendermint/tendermint/libs/strings"
+)
+
+const maxBypassMinFeeMsgGasUsage = uint64(200_000)
```

Comment:
Should this be a config file?

Reply:
there is no config file, there is only a `const.go` file but not in this package.

Reply:
we can, but the const is not in app.toml, these are two different things. the app.toml only sets the bypass fee types.

Reply:
@yaruwangway Do you think it would make sense for us to simply add a step to update a config file when a new field has been added with a default value?
Essentially a migration to the config files similar to how we do it for a migration to a new schema?

[tokamak](https://app.slack.com/team/U0310FRK1U4)
  [1 hour ago](https://interchainfoundation.slack.com/archives/C02QDK3JSJK/p1656334528805619?thread_ts=1656333756.169929&cid=C02QDK3JSJK)
we could always create a backup of the exiting app.toml

Reply:
Plus my assumption is that this field will become part of the config recommendations moving forward rite?

Reply:
At minimum we should write up a tech debt ticket to pull out all variables like this into some local const/vars file, for each directory

Reply:
Config migration is an alternative solution but since we have the fallback option working going to resolve this conversation.

This feature will eventually be added to the Cosmos SDK where [config validation](https://github.dev/cosmos/cosmos-sdk/blob/3283617cfb14dc90bf93e409a47bd1e37cd1f309/server/config/config.go#L363) can take place.